src/mito2/src/memtable/time_series.rs

```diff
@@ -237,39 +247,44 @@ impl Series {
         self.freeze(region_metadata);
 
         let mut frozen = self.frozen.clone();
-        let values = if frozen.len() == 1 {
+        let values = if frozen.is_empty() {
+            Values::new_empty(region_metadata)
+        } else if frozen.len() == 1 {
             frozen.pop().unwrap()
         } else {
             // TODO(hl): We should keep track of min/max timestamps for each values and avoid
             // cloning and sorting when values do not overlap with each other.
 
-            let total_len: usize = frozen.iter().map(|v| v.timestamp.len()).sum();
-            let mut builder = ValueBuilder::new(region_metadata, total_len);
-
-            for v in frozen {
-                let len = v.timestamp.len();
-                builder
-                    .timestamp
-                    .extend_slice_of(&*v.timestamp, 0, len)
-                    .context(CompactValuesSnafu)?;
-                builder
-                    .sequence
-                    .extend_slice_of(&*v.sequence, 0, len)
-                    .context(CompactValuesSnafu)?;
-
-                builder
-                    .op_type
-                    .extend_slice_of(&*v.op_type, 0, len)
-                    .context(CompactValuesSnafu)?;
-
-                for (idx, f) in v.fields.iter().enumerate() {
-                    builder.fields[idx]
-                        .extend_slice_of(&**f, 0, len)
-                        .context(CompactValuesSnafu)?;
-                }
+            let column_size = frozen.get(0).unwrap().fields.len() + 3;
```

Comment:
Same as `get(0).unwrap()`
```suggestion
            let column_size = frozen[0].fields.len() + 3;
```